### ISSUE TITLE
Type of getCredentials is incorrect

### ISSUE THREAD

**Issue by objectiveSee** [auth0/react-native-auth0#734]
### Checklist

- [X] The issue can be reproduced in the [react-native-auth0 sample app](https://github.com/auth0-samples/auth0-react-native-sample) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/react-native-auth0#readme), [Examples](https://github.com/auth0/react-native-auth0/blob/master/EXAMPLES.md), and [FAQ](https://github.com/auth0/react-native-auth0/blob/master/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/react-native-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/react-native-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

The function `getCredentials` specifies that it returns a promise with type `Credentials`. However, if the user is not logged in when you call the function then `undefined` is returned

```
const getCredentials: (scope?: string, minTtl?: number, parameters?: Record<string, unknown>, forceRefresh?: boolean) => Promise<Credentials>
```

### Reproduction

1. User is not logged in
2. Call `getCredentials` and log it's return value. You will set it's undefined

```
  useEffect(() => {
    const loadApiAccessToken = async () => {
      try {
        const credentials = await getCredentials() // Auth0 lets you retrieve credentials from secure storage
        console.log(`creds are: ${JSON.stringify(credentials)}`).  // 🐛BUG! This shouldn't be undefined
        const { accessToken } = credentials
        if (accessToken) {
          setApiAccessToken(accessToken);
        }
      } catch (error) {
        console.log('Auth0: Error loading accessToken', error)
      }
    }

    loadApiAccessToken()
  }, [])
```

### Additional context

Solution: Update the types on getCredentials so that `Credentials` is optional or allows for `null`/`undefined`, or return a credentials object with no accessToken (eg. accessToken is optional).  

### react-native-auth0 version

3.0.0

### React Native version

0.72.4

### Expo version

49.0.7

### Platform

iOS

### Platform version(s)

iOS Simulator 16.4

**Comment by poovamraj** [comment]
@objectiveSee Agree with you on that. Thanks for the callout. We will fix this in the next release.

**Comment by poovamraj** [comment]
@objectiveSee I am not sure why you are facing the issue because we have defined `getCredentials` this way

```
  getCredentials: (
    scope?: string,
    minTtl?: number,
    parameters?: Record<string, unknown>,
    forceRefresh?: boolean
  ) => Promise<Credentials | undefined>;
  ```
  
As you can see here - https://github.com/auth0/react-native-auth0/blob/master/src/hooks/auth0-context.ts#L104

Can you explain this in more detail?

**Comment by objectiveSee** [comment]
@poovamraj I see in the documentation that it's returning `Promise<Credentials | undefined>` but when I hover over the function in our code I see `Promise< Credentials>`. Would this be a VS Code issue? I am not sure where this documentation reads from when you hover on a function. It seems like Typescript is thinking that credentials will always be defined in terms of our type checking.

![Screenshot 2023-09-28 at 8 43 41 AM](https://github.com/auth0/react-native-auth0/assets/1684106/edb5655c-4072-4c4c-b4e3-9fb798abcc23)
![Screenshot 2023-09-28 at 8 46 44 AM](https://github.com/auth0/react-native-auth0/assets/1684106/8beb6bf5-2c46-48ef-ac2c-b2adc6e66165)

**Comment by objectiveSee** [comment]
@poovamraj we are using Auth0 library 3.0.1. I am seeing this when I cmd+click on the function:

![Screenshot 2023-09-28 at 8 50 17 AM](https://github.com/auth0/react-native-auth0/assets/1684106/bfd70386-f61c-42f1-87c3-55c2a6adc1a7)

But then when I hover on the function, typescript isn't picking up on the `undefined` value:

![Screenshot 2023-09-28 at 8 43 41 AM](https://github.com/auth0/react-native-auth0/assets/1684106/abfdad96-18d6-49f8-8dfd-e7fdc4397027)

**Comment by poovamraj** [comment]
Have you installed any types for RNA? This could be from the community contributed types which are obselete now since we provide the types through our library itself

**Comment by objectiveSee** [comment]
No, we are just using `"react-native-auth0": "3.0.1",` without any community things. The only other auth-related library we use is `"aws-appsync-auth-link`. I also looked at `yarn.lock` and confirmed there's just one Auth0 library being used.

**Comment by poovamraj** [comment]
@objectiveSee I am not able to see this in our sample repo [here](https://github.com/auth0-samples/auth0-react-native-sample/tree/master/00-Login-Hooks). This looks like issue in VSCode and not in the library. We will close this issue for now but please comment here if you can reproduce this in our sample repo.

<img width="1840" alt="Screenshot 2023-10-03 at 11 41 37" src="https://github.com/auth0/react-native-auth0/assets/15910425/6dc4e972-a96d-4c43-86a3-0141919e6eed">

**Comment by objectiveSee** [comment]
Thank you for looking into this. I will try to figure out what is going on within VS Code 🧑‍💻 🤞
